subsystems/loggingsubsystem.py

```diff
@@ -0,0 +1,50 @@
+from functools import reduce
+from operator import add
+
+from commands2 import SubsystemBase
+from ntcore import NetworkTableInstance
+from wpilib import PowerDistribution, SmartDashboard
+
+from operatorinterface import OperatorInterface
+
+import constants
+
+
+class LoggingSubsystem(SubsystemBase):
+    def __init__(self, oi: OperatorInterface) -> None:
+        SubsystemBase.__init__(self)
+        self.setName(__class__.__name__)
+
+        self.pdh = PowerDistribution(1, PowerDistribution.ModuleType.kRev)
+        self.oi = oi
+        self.dsTable = NetworkTableInstance.getDefault().getTable(
+            constants.kJoystickKeyLogPrefix
+        )
+
+    def periodic(self) -> None:
+        SmartDashboard.putData(self.pdh)
+
+        for controller in self.oi.controllers.values():
```

Comment:
What is the point of all this

Reply:
this is a pythonic implementation of controllers being logged to the driver station log, and consequentially I used the format that advantagekit uses for advantagescope for pushing joysticks

Reply:
https://github.com/Mechanical-Advantage/AdvantageScope/blob/main/docs/tabs/JOYSTICKS.md